api/envoy/config/core/v3/proxy_protocol.proto

```diff
@@ -47,4 +56,10 @@ message ProxyProtocolConfig {
   // This config controls which TLVs can be passed to upstream if it is Proxy Protocol
   // V2 header. If there is no setting for this field, no TLVs will be passed through.
   ProxyProtocolPassThroughTLVs pass_through_tlvs = 2;
+
+  // Custom TLV metadata to be sent in the upstream PROXY protocol header. This field
+  // is only valid when the version is V2. These TLVs can be defined both in the
+  // endpoint host's typed metadata and at the config-level. In the case that the same
+  // TLV is defined in both places, the endpoint host's typed metadata will take precedence.
+  repeated TlvEntry custom_tlvs = 3;
```

Comment:
What about `added_tlvs` for the name of this field? I think that complements `pass_through_tlvs` better.

Reply:
SGTM. Done in https://github.com/envoyproxy/envoy/pull/37591/commits/1c02dc137b56a1131cd815accdb62cec2a4e55c7.